orc/orcrules-neon.c

```diff
@@ -845,85 +1092,178 @@ neon_rule_loadX (OrcCompiler *compiler, void *user, OrcInstruction *insn)
     ptr_register = src->ptr_register;
   }
 
-  if (size >= 8) {
-    if (is_aligned) {
-      if (size == 32) {
-        ORC_ASM_CODE(compiler,"  vld1.64 { %s, %s, %s, %s }, [%s,:256]%s\n",
-            orc_neon_reg_name (dest->alloc),
-            orc_neon_reg_name (dest->alloc + 1),
-            orc_neon_reg_name (dest->alloc + 2),
-            orc_neon_reg_name (dest->alloc + 3),
-            orc_arm_reg_name (ptr_register),
-            update ? "!" : "");
-        code = 0xf42002dd;
-      } else if (size == 16) {
-        ORC_ASM_CODE(compiler,"  vld1.64 { %s, %s }, [%s,:128]%s\n",
-            orc_neon_reg_name (dest->alloc),
-            orc_neon_reg_name (dest->alloc + 1),
-            orc_arm_reg_name (ptr_register),
-            update ? "!" : "");
-        code = 0xf4200aed;
-      } else if (size == 8) {
-        ORC_ASM_CODE(compiler,"  vld1.64 %s, [%s]%s\n",
-            orc_neon_reg_name (dest->alloc),
-            orc_arm_reg_name (ptr_register),
-            update ? "!" : "");
-        code = 0xf42007cd;
+  if (compiler->is_64bit) {
```

Comment:
Cyclomatic Complexity (CC) is going to go wild. Be careful on the logic. (or try to refactor it with simpler logic or static functions for repeated tasks)

Reply:
Yes, I will consider resolving this problem.
Actually, I was trying to follow the format similar to the original code, but yes, we don't need to do.